admin_ui/components/Input.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Input = ({
```

Comment:
@chaidas let me briefly explain how I'd write the <Input /> component...

For start, I wouldn't add any label / error logic in here. Why? Because I would want to be able to use this component separately, i.e. outside a typical form.

If you do this then you don't need to define every attribute, but instead use the handy `...otherProps` syntax, e.g.

```javascript
import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ className, ...otherProps }) => {
  return (
      <input
        {...otherProps}
        className={`block border border-grey p-2 rounded leading-normal ${className}`}
      />
  );
};

Input.propTypes = {
  className: PropTypes.string,
};

export default Input;
```

Hope that makes sense.

Reply:
Then the question is, if you remove the label / error logic from the input component, where do you put it? I would suggest creating a `<FormField />` component that accepts label/error and children, so you can pass any kind of form input, e.g. text-input, select-box, color-picker, etc.